From 16b9253c25ce177fc4927b29b47abaf69c4062f3 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 23 Jun 2016 23:46:45 +0000
Subject: [PATCH] default disable challenge response

---
 servconf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servconf.c b/servconf.c
index 611668e..52ae47c 100644
--- a/servconf.c
+++ b/servconf.c
@@ -296,7 +296,7 @@ fill_default_server_options(ServerOptions *options)
 	if (options->kbd_interactive_authentication == -1)
 		options->kbd_interactive_authentication = 0;
 	if (options->challenge_response_authentication == -1)
-		options->challenge_response_authentication = 1;
+		options->challenge_response_authentication = 0;
 	if (options->permit_empty_passwd == -1)
 		options->permit_empty_passwd = 0;
 	if (options->permit_user_env == -1)
-- 
2.9.0

