From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 7 Nov 2018 14:38:46 -0800
Subject: [PATCH] Stateless moduli.

---
 dh.c        | 2 +-
 pathnames.h | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/dh.c b/dh.c
index b5bb35e..2a2176f 100644
--- a/dh.c
+++ b/dh.c
@@ -152,7 +152,7 @@ choose_dh(int min, int wantbits, int max)
 	int best, bestcount, which, linenum;
 	struct dhgroup dhg;
 
-	if ((f = fopen(_PATH_DH_MODULI, "r")) == NULL) {
+	if ((f = fopen(_PATH_DH_MODULI, "r")) == NULL && (f = fopen(_PATH_DH_MODULI_STATELESS, "r")) == NULL) {
 		logit("WARNING: could not open %s (%s), using fixed modulus",
 		    _PATH_DH_MODULI, strerror(errno));
 		return (dh_new_group_fallback(max));
diff --git a/pathnames.h b/pathnames.h
index f7ca5a7..af527f2 100644
--- a/pathnames.h
+++ b/pathnames.h
@@ -42,6 +42,8 @@
 #define _PATH_HOST_XMSS_KEY_FILE	SSHDIR "/ssh_host_xmss_key"
 #define _PATH_HOST_RSA_KEY_FILE		SSHDIR "/ssh_host_rsa_key"
 #define _PATH_DH_MODULI			SSHDIR "/moduli"
+/* Stateless moduli fallback */
+#define _PATH_DH_MODULI_STATELESS      	"/usr/share/defaults/ssh/moduli"
 
 #ifndef _PATH_SSH_PROGRAM
 #define _PATH_SSH_PROGRAM		"/usr/bin/ssh"
