--- openssh-6.7p1/cipher.c~	2015-03-09 17:20:11.000000000 -0400
+++ openssh-6.7p1/cipher.c	2015-03-09 17:24:53.873984891 -0400
@@ -616,12 +616,6 @@
 	const struct sshcipher *c = cc->cipher;
 	int plen = 0;
 
-	if (c->evptype == EVP_rc4) {
-		plen = EVP_X_STATE_LEN(cc->evp);
-		if (dat == NULL)
-			return (plen);
-		memcpy(dat, EVP_X_STATE(cc->evp), plen);
-	}
 	return (plen);
 #else
 	return 0;
@@ -635,9 +629,5 @@
 	const struct sshcipher *c = cc->cipher;
 	int plen;
 
-	if (c->evptype == EVP_rc4) {
-		plen = EVP_X_STATE_LEN(cc->evp);
-		memcpy(EVP_X_STATE(cc->evp), dat, plen);
-	}
 #endif
 }
