Index: openssh-6.7p1/servconf.c
===================================================================
--- openssh-6.7p1.orig/servconf.c	2015-03-03 17:23:53.687206411 +0000
+++ openssh-6.7p1/servconf.c	2015-03-03 17:25:24.775208550 +0000
@@ -201,7 +201,7 @@
 	if (options->key_regeneration_time == -1)
 		options->key_regeneration_time = 3600;
 	if (options->permit_root_login == PERMIT_NOT_SET)
-		options->permit_root_login = PERMIT_YES;
+		options->permit_root_login = PERMIT_NO;
 	if (options->ignore_rhosts == -1)
 		options->ignore_rhosts = 1;
 	if (options->ignore_user_known_hosts == -1)
