--- openssh-9.0p1/pathnames.h~	2022-04-06 00:47:48.000000000 +0000
+++ openssh-9.0p1/pathnames.h	2022-09-13 14:09:50.029014431 +0000
@@ -35,6 +35,7 @@
  * should be world-readable.
  */
 #define _PATH_SERVER_CONFIG_FILE	SSHDIR "/sshd_config"
+#define _PATH_SERVER_CONFIG_FILE_STATELESS "/usr/share/defaults/ssh/sshd_config"
 #define _PATH_HOST_CONFIG_FILE		SSHDIR "/ssh_config"
 #define _PATH_HOST_DSA_KEY_FILE		SSHDIR "/ssh_host_dsa_key"
 #define _PATH_HOST_ECDSA_KEY_FILE	SSHDIR "/ssh_host_ecdsa_key"
--- openssh-9.8p1/sshd.c~	2024-07-01 04:36:28.000000000 +0000
+++ openssh-9.8p1/sshd.c	2024-07-01 14:33:45.204747942 +0000
@@ -1400,8 +1400,14 @@
 	/* Fetch our configuration */
 	if ((cfg = sshbuf_new()) == NULL)
 		fatal("sshbuf_new config failed");
-	if (strcasecmp(config_file_name, "none") != 0)
-		load_server_config(config_file_name, cfg);
+	if (strcasecmp(config_file_name, "none") != 0) {
+               if (access(config_file_name, R_OK) != 0 && strcmp(config_file_name, _PATH_SERVER_CONFIG_FILE) == 0) {
+                       load_server_config(_PATH_SERVER_CONFIG_FILE_STATELESS, cfg);
+               } else {
+                       load_server_config(config_file_name, cfg);
+               }
+       }
+
 
 	parse_server_config(&options, config_file_name, cfg,
 	    &includes, NULL, 0);
